src/test/java/org/opensearch/neuralsearch/sparse/algorithm/BatchClusteringTaskTests.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.sparse.algorithm;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.lucene.index.MergeState;
+import org.apache.lucene.index.FieldInfo;
+import org.apache.lucene.util.BytesRef;
+import org.junit.Before;
+import org.mockito.MockitoAnnotations;
+import org.opensearch.neuralsearch.sparse.AbstractSparseTestBase;
+import org.opensearch.neuralsearch.sparse.common.InMemoryKey;
+import org.opensearch.neuralsearch.sparse.testsPrepareUtils;
+
+import java.util.Arrays;
+import java.util.List;
+import java.lang.reflect.Field;
+
+public class BatchClusteringTaskTests extends AbstractSparseTestBase {
+    private List<BytesRef> terms;
+    private InMemoryKey.IndexKey key;
+
+    @Before
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        MockitoAnnotations.openMocks(this);
+
+        terms = Arrays.asList(new BytesRef("term1"), new BytesRef("term2"));
+        key = new InMemoryKey.IndexKey(null, "test_field");
+    }
+
+    public void testConstructorDeepCopiesTerms() throws Exception {
+        // Setup
+        List<BytesRef> originalTerms = Arrays.asList(new BytesRef("term1"), new BytesRef("term2"));
+
+        // Execute - create task with null mergeState to test constructor
+        BatchClusteringTask task = new BatchClusteringTask(originalTerms, key, 0.5f, 0.3f, 10, null, null);
+
+        // Verify task is created
+        assertNotNull("Task should be created successfully", task);
+
+        Field termsField = BatchClusteringTask.class.getDeclaredField("terms");
+        termsField.setAccessible(true);
+        @SuppressWarnings("unchecked")
+        List<BytesRef> taskTerms = (List<BytesRef>) termsField.get(task);
+
+        // Verify deep copy by checking actual content
+        assertEquals("First term should be 'term1'", "term1", taskTerms.get(0).utf8ToString());
+
+        // Modify original terms to verify deep copy
+        originalTerms.get(0).bytes[0] = (byte) 'X';
+
+        // Verify task's terms remain unchanged (proving deep copy worked)
+        assertEquals("Task's first term should still be 'term1'", "term1", taskTerms.get(0).utf8ToString());
+        assertNotEquals("Original term should now be different", "term1", originalTerms.get(0).utf8ToString());
+    }
+
+    public void testTaskCreationWithDifferentParameters() throws Exception {
```

Comment:
this test doesn't seem to bring much value as one value is not different comparing to another value when it's simply assignment. It makes more sense if you test null terms and empty terms.